Core/include/Acts/Propagator/AtlasStepper.hpp

```diff
@@ -304,7 +300,18 @@ class AtlasStepper {
     size_t debugMsgWidth = 50;
   };
 
-  AtlasStepper(bfield_t bField) : m_bField(std::move(bField)){};
+  AtlasStepper(std::shared_ptr<const MagneticFieldProvider> bField)
+      : m_bField(std::move(bField)){};
+
+  template <typename charge_t>
```

Comment:
Ok, I see, this will then need a Free/Bound parameter constructor for #727 - which is why I wanted this to go in first.